nestedtensor/csrc/UnaryOps.cpp

```diff
@@ -0,0 +1,235 @@
+                               const Tensor& self,
+                               optional<Scalar> min,
+                               optional<Scalar> max) {
+  auto result_impl = get_nested_tensor_impl(result);
+  auto self_impl = get_nested_tensor_impl(self);
+  apply([min, max](at::Tensor& result, const at::Tensor tensor)
+          { return at::clamp_out(result, tensor, min, max); },
+      result_impl->_data.get_structure(),
+      self_impl->_data.get_structure());
+  return result;
+}
+
+Tensor& NestedTensor_clamp_min_(Tensor& self, Scalar min) {
+  auto self_impl = get_nested_tensor_impl(self);
+  auto f = [min](at::Tensor& tensor) { at::clamp_min_(tensor, min); };
+  apply<decltype(f)>(std::move(f), self_impl->_data.get_structure());
+  return self;
+}
+
+Tensor NestedTensor_clamp_min(const Tensor& self, Scalar min) {
+  auto self_impl = get_nested_tensor_impl(self);
+  return at::detail::make_tensor<NestedTensorImpl>(
+      map([min](at::Tensor tensor) { return at::clamp_min(tensor, min); },
+          self_impl->_data.get_structure()));
+}
+
+Tensor& NestedTensor_clamp_min_out(Tensor& result,
+                               const Tensor& self,
+                               Scalar min) {
+  auto result_impl = get_nested_tensor_impl(result);
+  auto self_impl = get_nested_tensor_impl(self);
+  apply([min](at::Tensor& result, const at::Tensor tensor)
+          { return at::clamp_min_out(result, tensor, min); },
+      result_impl->_data.get_structure(),
+      self_impl->_data.get_structure());
+  return result;
+}
+
+Tensor& NestedTensor_clamp_max_(Tensor& self, Scalar min) {
+  auto self_impl = get_nested_tensor_impl(self);
+  auto f = [min](at::Tensor& tensor) { at::clamp_max_(tensor, min); };
+  apply<decltype(f)>(std::move(f), self_impl->_data.get_structure());
+  return self;
+}
+
+Tensor NestedTensor_clamp_max(const Tensor& self, Scalar min) {
+  auto self_impl = get_nested_tensor_impl(self);
+  return at::detail::make_tensor<NestedTensorImpl>(
+      map([min](at::Tensor tensor) { return at::clamp_max(tensor, min); },
+          self_impl->_data.get_structure()));
+}
+
+Tensor& NestedTensor_clamp_max_out(Tensor& result,
+                               const Tensor& self,
+                               Scalar min) {
+  auto result_impl = get_nested_tensor_impl(result);
+  auto self_impl = get_nested_tensor_impl(self);
+  apply([min](at::Tensor& result, const at::Tensor tensor)
+          { return at::clamp_max_out(result, tensor, min); },
+      result_impl->_data.get_structure(),
+      self_impl->_data.get_structure());
+  return result;
+}
+
+// NOTE: Missing at::lgamma_ -> very annoying
+Tensor& NestedTensor_lgamma_(Tensor& self) {
+  auto self_impl = get_nested_tensor_impl(self);
+  auto f = [](at::Tensor& tensor) { tensor.lgamma_(); };
+  apply<decltype(f)>(std::move(f), self_impl->_data.get_structure());
+  return self;
+}
+
+// NOTE: Missing at::erfinv_ -> very annoying
```

Comment:
Correct me if I'm wrong, but it looks like the problem here is that `NestedTensor_unary_` only takes function pointers, so you can't pass it something like `Tensor::erfinv_`.

The way I worked around it with BatchedTensor is: you can actually write a template function that takes in "pointers-to-methods". Something like the following, modulo syntax (I did not run it through a compiler):

```
template <class F, F method>
Tensor& NestedTensor_unary_method_(Tensor& self) {
  auto self_impl = get_nested_tensor_impl(self);
  auto f = [](at::Tensor& tensor) { tensor.*method(); };
  apply<decltype(f)>(std::move(f), self_impl->_data.get_structure());
  return self;
}
```

Reply:
Ah I wasn't aware of this syntax! Still annoying that we have to write a separate function for inplace instead of a single UNARY_OP as it currently is.